specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2018-02-14/keyvault.json

```diff
@@ -665,6 +665,187 @@
           "application/json"
         ]
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}/privateEndpointConnections/{privateEndpointConnectionName}": {
+      "get": {
+        "tags": [
+          "PrivateEndpointConnections"
+        ],
+        "operationId": "PrivateEndpointConnections_Get",
+        "description": "Gets the specified private endpoint connection associated with the key vault.",
+        "x-ms-examples": {
+          "KeyVaultGetPrivateEndpointConnection": {
+            "$ref": "./examples/getPrivateEndpointConnection.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/VaultName"
+          },
+          {
+            "$ref": "#/parameters/PrivateEndpointConnectionName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Private endpoint connection successfully returned.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnection"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "PrivateEndpointConnections"
+        ],
+        "operationId": "PrivateEndpointConnections_Put",
+        "description": "Updates the specified private endpoint connection associated with the key vault.",
+        "x-ms-examples": {
+          "KeyVaultPutPrivateEndpointConnection": {
+            "$ref": "./examples/putPrivateEndpointConnection.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/VaultName"
+          },
+          {
+            "$ref": "#/parameters/PrivateEndpointConnectionName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "properties",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnection"
+            },
+            "description": "The intended state of private endpoint connection."
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
>200 [](start = 11, length = 3)

Aren't you missing a 201 response status code? Creation should be 201 in general and one of the accepted patterns for long running operations uses 201+provisioningState.

Reply:
Same thing. The `storage.json` does not specify `201`.

Reply:
What is critical here is that the swagger matches the server behavior. The way this is described, the operation is always synchronous. If that is correct, no change is needed. If it is async, then a 201 response and x-ms-long-running-operation is the preferred async pattern.

Reply:
@majastrz  The PUT here is never used for Creation. The server never returns 201. BTW, I do agree that PUT was a poor choice of method. It should be POST, but when this was raised, other teams were already using PUT.

@KrisBash Regarding `x-ms-long-running-operation`, please see https://github.com/Azure/azure-rest-api-specs/pull/7631#discussion_r363998777.

Reply:
PUT is required for template deployments.. POST is not idempotent. So, template deployments use only PUT. 

---
In reply to: [363999344](https://github.com/Azure/azure-rest-api-specs/pull/7631#discussion_r363999344) [](ancestors = 363999344)

Reply:
Yes. This API is not used for template deployment.